XiEditor/StatusBar.swift

```diff
@@ -210,9 +212,20 @@ class StatusBar: NSView {
         self.needsDisplay = true
     }
 
-    // Hides the status bar if there is no item currently.
-    func checkStatusBarVisibility() {
-        self.isHidden = currentItems.isEmpty
+    // Shows/hides the statusbar based on whether it has any contents
+    func updateStatusBarVisibility() {
+        if !currentItems.isEmpty {
+            self.hideTimer?.invalidate()
+            self.isHidden = false
+        } else if !self.isHidden {
+            // Wait a moment before hiding the bar to avoid blinking in the case
+            // that a single item is added and removed repeatedly in rapid succession.
+            if #available(OSX 10.12, *) {
+                hideTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) {_ in self.isHidden = true}
```

Comment:
The block closure at the end should have some spaces to be easier to read, so should look like  `{ _ in self.isHidden = true }`